docs/src/architecture/state.md

```diff
@@ -11,89 +11,60 @@ the system maintains 3 databases to describe the state:
-
-It is important to note that the decision when to discard full account states does not need to be a part of the protocol - every node could decide for themselves. For example, there could be nodes which prune full state very aggressively, and in effect, they would only be able to include private transactions in their blocks. There could also be nodes which decide to keep full account states for years - so that they could execute transactions which few other nodes could (presumably for a higher fee).
-
-This approach eliminates the need for complicated mechanisms such as state rent. The longer a public account remains unused, the fewer nodes would want to keep its full state. The fewer nodes keep its full, state, the higher fees can be demanded for executing transactions against this account. Thus, the nodes which chose to keep full states longer should get naturally compensated for their services.
-
-### Note database
-
-Notes could be recorded in an append-only accumulator similar to the one described [here](https://eprint.iacr.org/2015/718). Using such an accumulator is important for two reasons:
+Notes are recorded in an append-only accumulator, a [Merkle Mountain Range](https://github.com/opentimestamps/opentimestamps-server/blob/master/doc/merkle-mountain-range.md). This is important for two reasons:
 
 1. Membership witnesses against such an accumulator needs to be updated very infrequently.
 2. Old membership witnesses can be extended to be used with a new accumulator value, but this extension does not need to be done by the original witness holder.
 
 Both of these properties are needed for supporting local transactions and private accounts.
 
-Notes database could look as shown on the diagram below. Here, the database contains $6$ notes: $1$ through $6$, and the commitment to this database are the roots of individual trees `(a, b)`. Thus, the size of the commitment grows logarithmically with the number of items in it.
+There are two types of [notes](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html):
+* **Public notes** where the entire note content is recorded in the state. 
+* **Private notes** where only a note's hash is recorded in the state. 
+
+As with accounts, there is a strong incentive to use private notes as they result in lower fees. This is also beneficial to the network as a private note adds only 64 bytes to the state (32 bytes when it is produced, and 32 bytes when it is consumed).
+
+Notes database look as shown on the diagram below. Each leaf is a block header which contains the commitment to all notes created in that block. Note, the size of the commitment grows logarithmically with the number of items in it.
```

Comment:
"Note, the size of the commitment grows logarithmically with the number of items in it." I'm not sure this is accurate.  The size of the commitment is constant - a single hash digest / word. Did you mean the size / depth of the data structure?

Reply:
good catch